test/unit/UnzerTest.php

```diff
@@ -222,6 +222,20 @@ public function unzerShouldForwardCancelActionCallsToTheCancelService(
         $unzer->$unzerMethod(...$unzerParams);
     }
 
+    /**
+     * Verify unzer provides correct environment based on the set key.
+     *
+     * @test
+     */
+    public function unzerShouldProvideExpectedKeyEnvironment()
+    {
+        $unzerProduction = new Unzer('p-priv-abcdef-g');
+        $this->assertEquals(true, $unzerProduction->hasProductionKey());
+
+        $unzerSandbox = new Unzer('s-priv-abcdef-g');
+        $this->assertEquals(false, $unzerSandbox->hasProductionKey());
```

Comment:
I actually removed this test now, since it got moved into a different class and I changed it to private.
It is tested indirectly by the test "environmentUrlSwitchesWithEnvironmentVariable".